### PR TITLE
SITES-19431: GA UX Improvements

### PR DESCRIPTION
**Pull request by irenelagno**
- SITES-19432: Generation Variation updates
- SITES-20317: change icon for re-use and add label with tool tip

## Description

See changes on qa: https://experience-qa.adobe.com/?devMode=true&shell_source=workspace&workspace=iraDev&shell_ims=prod#/aem/generate-variations/

## How Has This Been Tested?

- test UI with different number of generated variations >3 and <3
- need to test that integration of CF editor with Generate Variation is work as expected from the UI point of view 
- 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by askayastha22 — CHANGES_REQUESTED** [review]
- The left and right arrow buttons don't look like the mocks. Notice the subtle shadow effect. We can leverage the icons available in [`S2 / Icons Global`](https://www.figma.com/design/3gnNHtdUpqqP0evNF0XQIW/S2-%2F-Icons-Global?node-id=1310-23&m=dev) in Figma.
Screenshot 2024-05-09 at 12 30 20 PM
Screenshot 2024-05-09 at 12 31 12 PM

- The arrow buttons overlay on top of the Prompt Editor.
Screenshot 2024-05-09 at 12 28 53 PM

- Left and right arrow buttons have different amount of spacing from the Result Card border.
Screenshot 2024-05-09 at 12 28 29 PM

- Notice how the persistence of the arrow buttons blocks the variant text. I propose the arrow buttons to appear only when the user hovers over the Variant card area. I have already implemented this when hovering over the image thumbnails. Making the buttons appear only when hovering over the text variant will be a consistent behavior, like when the buttons appear when hovering over thumbnail images.
Screenshot 2024-05-09 at 12 45 46 PM

**Comment by irenelagno** [comment]
@askayastha22  your suggestions are fixed. Please check

**Review by lenaorobei — APPROVED** [review]

**Review by askayastha22 — CHANGES_REQUESTED** [review]
Some minor issues:

- Arrow buttons are not aligned vertically with each other.
- Arrow buttons are not centered vertically within the variant container.
Screenshot 2024-05-10 at 3 59 22 PM

File: web-src/src/index.css [hunk 1]
```diff
@@ -133,4 +133,56 @@ body {
...
+  z-index: 2;
+  border-radius: 50%;
+  width: auto;
+  height: auto;
+  filter: opacity(0);
+}
+
+.slick-slider:hover .slick-arrow {
+  filter: opacity(1);
+}
```
Comment (askayastha22) [inline web-src/src/index.css:164]: Can we slightly ease the hover effect by transitioning the buttons from opacity 0 to 1?
`transition: opacity 0.3s ease-in-out 0.1s;`

Comment (irenelagno) [reply web-src/src/index.css]: @askayastha22  It should be fine now :-)
Screen Shot 2024-05-10 at 5 04 02 PM

File: web-src/src/components/PromptResultCard.js [hunk 2]
```diff
@@ -211,6 +208,15 @@ export function PromptResultCard({ result, ...props }) {
       });
   }, [result, firefallService]);
 
+  const formattedTimestamp = new Date(result.timestamp).toLocaleString('en-US', {
```
Comment (askayastha22) [inline web-src/src/components/PromptResultCard.js:211]: You can get the user's set locale from ShellProvider:
```
import { useShellContext } from './ShellProvider.js';

const { user } = useShellContext();
...
const formattedTimestamp = new Date(result.timestamp).toLocaleString(user.locale, {
```

**Comment by irenelagno** [comment]
@askayastha22 fixed your comments

File: web-src/src/index.css [hunk 3]
```diff
@@ -133,4 +133,57 @@ body {
...
+  z-index: 2;
+  border-radius: 50%;
+  width: auto;
+  height: auto;
+  filter: opacity(0);
+}
+
+.slick-slider:hover .slick-arrow {
+  filter: opacity(1);
+  transition: opacity 0.3s ease-in-out 0.1s
```
Comment (askayastha22) [inline web-src/src/index.css:164]: Sorry for being nitpicky.
- I didn't see this transition taking effect in the feature QA URL.
- I played around with the CSS and found this to work.
```suggestion
  opacity: 1;
```

File: web-src/src/index.css [hunk 4]
```diff
@@ -133,4 +133,57 @@ body {
...
+  background-size: contain;
+  background: url('./assets/variations-scroll-indicator.svg');
+}
+.slick-arrow {
+  background: #fff;
+  z-index: 2;
+  border-radius: 50%;
+  width: auto;
+  height: auto;
+  filter: opacity(0);
```
Comment (askayastha22) [inline web-src/src/index.css:159]: I found 0.2s transition time work best for the hovering effect.
```suggestion
  opacity: 0;
  transition: opacity 0.2s ease-in-out 0.1s;
```

**Review by askayastha22 — CHANGES_REQUESTED** [review]
Sorry for being nitpicky.

File: web-src/src/index.css (hunk 3 above)
Comment (irenelagno) [reply web-src/src/index.css:164]: fixed

File: web-src/src/index.css (hunk 4 above)
Comment (irenelagno) [reply web-src/src/index.css:159]: fixed

**Comment by irenelagno** [comment]
@askayastha22 the transition should be fixed :)

**Review by askayastha22 — APPROVED** [review]

**Pull request merged**